action.yml

```diff
@@ -0,0 +1,237 @@
+      with:
+        images: ${{ inputs.google_ar_image_name }}
+        tags: |
+          # SHA for all events
+          type=sha,format=short
+          # Long SHA for all events
+          type=sha,format=long
+        labels: |
+            org.opencontainers.image.title=${{ github.event.repository.name }}
+            org.opencontainers.image.description=${{ github.event.repository.description }}
+            org.opencontainers.image.vendor="Sentry"
+            org.opencontainers.image.revision=${{ github.pull_request.head.sha || github.sha }}
+            org.opencontainers.image.version=${{ github.pull_request.head.sha || github.sha }}
+            org.opencontainers.image.source=${{ github.server_url }}/${{ github.repository }}
+            org.opencontainers.image.url=${{ github.server_url }}/${{ github.repository }}
+
+
+    - name: Setup Google Cloud authentication
+      if: steps.setup.outputs.publish_to_gar == 'true'
+      uses: google-github-actions/auth@ba79af03959ebeac9769e648f473a284504d9193 # v2.1.10
+      with:
+        workload_identity_provider: ${{ inputs.google_workload_identity_provider }}
+        service_account: ${{ inputs.google_service_account }}
+
+    - name: Configure Docker for Google Artifact Registry
+      if: steps.setup.outputs.publish_to_gar == 'true'
+      shell: bash
+      run: |
+        gcloud auth configure-docker $(echo "${{ inputs.google_ar_image_name }}" | cut -d'/' -f1)
+
+    - name: Build and push to GAR
+      if: steps.setup.outputs.publish_to_gar == 'true'
+      uses: docker/build-push-action@263435318d21b8e681c14492fe198d362a7d2c83 # v6.18.0
+      with:
+        context: ${{ inputs.build_context }}
+        file: ${{ inputs.dockerfile_path }}
+        platforms: linux/${{ inputs.platform }}
+        build-args: ${{ inputs.build_args }}
+        target: ${{ inputs.build_target }}
+        tags: ${{ steps.meta-gar.outputs.tags }}
+        labels: ${{ steps.meta-gar.outputs.labels }}
+        cache-from: type=gha,scope=${{ inputs.platform }}
+        cache-to: type=gha,mode=max,scope=${{ inputs.platform }}
+        provenance: mode=max
+        sbom: true
+        push: true
+
+    - name: Build only (no push)
+      id: build-only
+      if: steps.setup.outputs.publish_to_ghcr == 'false' && steps.setup.outputs.publish_to_gar == 'false'
+      uses: docker/build-push-action@263435318d21b8e681c14492fe198d362a7d2c83 # v6.18.0
+      with:
+        context: ${{ inputs.build_context }}
+        file: ${{ inputs.dockerfile_path }}
+        platforms: linux/${{ inputs.platform }}
+        build-args: ${{ inputs.build_args }}
+        target: ${{ inputs.build_target }}
+        tags: build-${{ inputs.image_name }}:${{ github.pull_request.head.sha || github.sha }}
+        cache-from: type=gha,scope=${{ inputs.platform }}
+        cache-to: type=gha,mode=max,scope=${{ inputs.platform }}
+        provenance: mode=max
+        sbom: true
```

Comment:
```suggestion
```

No need for these to be set when we are just building and not pushing. In some cases, it will actually cause the job to fail depending on the local container registry being used.